HHS/OPRE-OPS
# OPS-1827 - Allow user to edit incumbent on existing agreements

## What changed

Changed the update method used by POST/PATCH /Agreements/{id} so that if the vendor or incumbent is changed in the request then the change will be saved to the database. Also, did some refactoring to combine the method used by when setting incumbent/vendor during creation of agreements to be the same method used when editing agreements. 

## Issue

[OPS-1827](https://github.com/HHS/OPRE-OPS/issues/1827)

## How to test

1. Select Agreements from the top of the page.
2. Click the first agreement in the list.
3. Click edit
4. Under 'Incumbent', change the name to either 'Vendor 2' (an existing vendor under our system), or some random name. 
5. After the update, the new incumbent name should be reflected on the Agreement page. 

## Definition of Done Checklist
